### PR TITLE
Add OctaveControl

### PR DESCRIPTION
**Pull request by soiferj**

### CODE REVIEW

File: src/components/octaveControl.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        background-color:black;
+        color:green;
+    }
+    &::selection{
+        background: transparent;
+    }
+`
+
+interface OctaveControlProps {
+    setOctave: (octave: any) => void,
```
Comment (nspilman) [inline src/components/octaveControl.tsx]: is this truly any? would this be a number?

File: src/components/octaveControl.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    curOctave: any
+}
+
+const minOctave = 0;
+const maxOctave = 8;
+
+function OctaveControl({setOctave, curOctave}: OctaveControlProps){
+    return (
+        <div id="OctaveControl">
+            <StyledOctaveControl
```
Comment (nspilman) [inline src/components/octaveControl.tsx]: Can you add className "lower-octave" for html readability purposes?

File: src/components/octaveControl.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                onClick={() => setOctave(curOctave - 1 < minOctave ? minOctave : curOctave - 1)}
+            >
+                Octave -
+            </StyledOctaveControl>
+            <StyledOctaveControl
```
Comment (nspilman) [inline src/components/octaveControl.tsx]: Same as above

File: src/components/octaveControl.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+const minOctave = 0;
+const maxOctave = 8;
+
+function OctaveControl({setOctave, curOctave}: OctaveControlProps){
+    return (
+        <div id="OctaveControl">
+            <StyledOctaveControl
+                onClick={() => setOctave(curOctave - 1 < minOctave ? minOctave : curOctave - 1)}
```
Comment (nspilman) [inline src/components/octaveControl.tsx]: can you abstract out a private method for increaseVolume() that does this? Purely for readability

File: src/components/octaveControl.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+            >
+                Octave -
+            </StyledOctaveControl>
+            <StyledOctaveControl
+                onClick={() => setOctave(curOctave + 1 > maxOctave ? maxOctave : curOctave + 1)}
```
Comment (nspilman) [inline src/components/octaveControl.tsx]: Same as above - can you abstract out a private method for decreaseOctave() that does this? Purely for readability

File: src/components/octaveControl.tsx (hunk 4 above)
Comment (nspilman) [reply src/components/octaveControl.tsx]: and by "volume" I mean "octave"

File: src/components/keyboard.tsx [hunk 6]
```diff
@@ -13,21 +14,27 @@ const audioController = new AudioContextWrapper();
 
 function Keyboard() {
     const [isClicked, setIsClicked] = useState(false);
-    const octave = 4;
+    const [octave, setOctave] = useState(4);
```
Comment (nspilman) [inline src/components/keyboard.tsx:17]: I think this is good for now, but I want to discuss architecture of the controller. I think it'd be solid to have an abstracted component responsible for all of these state objects, and with this component responsible solely for render the components and passing them the requisite info for them to get the information they need.

File: src/components/octaveControl.tsx (hunk 1 above)
Comment (nspilman) [reply src/components/octaveControl.tsx]: and by 'this' I mean 'octave'. Shouldn't this be setOctave: (octave : number) => void ?

File: src/components/octaveControl.tsx (hunk 3 above)
Comment (nspilman) [reply src/components/octaveControl.tsx]: Can you add className "increase-octave" for html readability purposes?

File: src/components/octaveControl.tsx (hunk 1 above)
Comment (soiferj) [reply src/components/octaveControl.tsx]: oh the keyword is "number", i was looking for int! updated

Comment (nspilman) [reply src/components/octaveControl.tsx]: TIL all numbers in TS are floats except for BigInts - https://www.typescriptlang.org/docs/handbook/basic-types.html#number

**Review by nspilman — APPROVED** [review]

**Pull request merged**